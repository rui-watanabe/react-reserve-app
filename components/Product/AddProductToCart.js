import React from 'react';
import { Input } from 'semantic-ui-react';
import { useRouter } from 'next/router';
import axios from 'axios';
import baseUrl from '../../utils/baseUrl';
import cookie from 'js-cookie';

function AddProductToCart({ user, productId }) {
  const [quantity, setQuantity] = React.useState(1);
  const router = useRouter();

  async function handleAddProductToCart(){
    const url = `${baseUrl}/api/cart`;
    const payload = { quantity, productId};
    const token = cookie.get('token');
    const headers = { headers: { Authorization: token } };
    await axios.put(url, payload, headers);
  }

  return(
    <Input
      type="number"
      min="1"
      placeholder="Quantity"
      value={quantity}
      onChange={event => setQuantity(Number(event.target.value))}
      action={user ? {
        color: "orange",
        content: "Add to Cart",
        icon: "plus cart",
        onClick: handleAddProductToCart
      } : {
        color: "blue",
        content: "Sign Up To Purchase",
        icon: "signup",
        onClick: () => router.push('/signup')
      }}
    />
  )
}

export default AddProductToCart;
