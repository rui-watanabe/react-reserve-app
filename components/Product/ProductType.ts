import { UserModelType } from '../../models/UserModel/UserType';

export interface ProductAttributesProps {
  description: string;
  _id: string;
  user: UserModelType;
}

export interface ProductSummaryProps {
  name: string;
  mediaUrl: string;
  _id: string;
  price: number;
  sku: string;
  user: UserModelType;
}

export interface AddProductToCartProps {
  productId: string;
  user: UserModelType;
}
