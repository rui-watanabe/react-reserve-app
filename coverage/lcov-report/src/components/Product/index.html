
<!doctype html>
<html lang="en">

<head>
    <title>Code coverage report for src/components/Product</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../prettify.css" />
    <link rel="stylesheet" href="../../../base.css" />
    <link rel="shortcut icon" type="image/x-icon" href="../../../favicon.png" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../sort-arrow-sprite.png);
        }
    </style>
</head>
    
<body>
<div class='wrapper'>
    <div class='pad1'>
        <h1><a href="../../../index.html">All files</a> src/components/Product</h1>
        <div class='clearfix'>
            
            <div class='fl pad1y space-right2'>
                <span class="strong">69.44% </span>
                <span class="quiet">Statements</span>
                <span class='fraction'>50/72</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">68.75% </span>
                <span class="quiet">Branches</span>
                <span class='fraction'>11/16</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">25% </span>
                <span class="quiet">Functions</span>
                <span class='fraction'>3/12</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">66.67% </span>
                <span class="quiet">Lines</span>
                <span class='fraction'>42/63</span>
            </div>
        
            
        </div>
        <p class="quiet">
            Press <em>n</em> or <em>j</em> to go to the next uncovered block, <em>b</em>, <em>p</em> or <em>k</em> for the previous block.
        </p>
    </div>
    <div class='status-line medium'></div>
    <div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file medium" data-value="AddProductToCart.tsx"><a href="AddProductToCart.tsx.html">AddProductToCart.tsx</a></td>
	<td data-value="57.89" class="pic medium">
	<div class="chart"><div class="cover-fill" style="width: 57%"></div><div class="cover-empty" style="width: 43%"></div></div>
	</td>
	<td data-value="57.89" class="pct medium">57.89%</td>
	<td data-value="38" class="abs medium">22/38</td>
	<td data-value="50" class="pct medium">50%</td>
	<td data-value="8" class="abs medium">4/8</td>
	<td data-value="14.29" class="pct low">14.29%</td>
	<td data-value="7" class="abs low">1/7</td>
	<td data-value="51.61" class="pct medium">51.61%</td>
	<td data-value="31" class="abs medium">16/31</td>
	</tr>

<tr>
	<td class="file medium" data-value="ProductAttributes.tsx"><a href="ProductAttributes.tsx.html">ProductAttributes.tsx</a></td>
	<td data-value="73.91" class="pic medium">
	<div class="chart"><div class="cover-fill" style="width: 73%"></div><div class="cover-empty" style="width: 27%"></div></div>
	</td>
	<td data-value="73.91" class="pct medium">73.91%</td>
	<td data-value="23" class="abs medium">17/23</td>
	<td data-value="87.5" class="pct high">87.5%</td>
	<td data-value="8" class="abs high">7/8</td>
	<td data-value="25" class="pct low">25%</td>
	<td data-value="4" class="abs low">1/4</td>
	<td data-value="71.43" class="pct medium">71.43%</td>
	<td data-value="21" class="abs medium">15/21</td>
	</tr>

<tr>
	<td class="file high" data-value="ProductSummary.tsx"><a href="ProductSummary.tsx.html">ProductSummary.tsx</a></td>
	<td data-value="100" class="pic high">
	<div class="chart"><div class="cover-fill cover-full" style="width: 100%"></div><div class="cover-empty" style="width: 0%"></div></div>
	</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="11" class="abs high">11/11</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="11" class="abs high">11/11</td>
	</tr>

</tbody>
</table>
</div>
                <div class='push'></div><!-- for sticky footer -->
            </div><!-- /wrapper -->
            <div class='footer quiet pad2 space-top1 center small'>
                Code coverage generated by
                <a href="https://istanbul.js.org/" target="_blank">istanbul</a>
                at Sun Sep 20 2020 17:42:36 GMT+0900 (GMT+09:00)
            </div>
        </div>
        <script src="../../../prettify.js"></script>
        <script>
            window.onload = function () {
                prettyPrint();
            };
        </script>
        <script src="../../../sorter.js"></script>
        <script src="../../../block-navigation.js"></script>
    </body>
</html>
    