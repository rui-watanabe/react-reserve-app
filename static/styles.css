:root {
  --bodyBg: #f0f4f8;
  --menuBg: #008080;
  --progressBar: #54d1db;
}

body {
  background: var(--bodyBg);
}

#menu {
  background: var(--menuBg);
}

/* Overrides default styled accordion styles (too much bottom padding) */
.ui.styled.accordion .content {
  margin: 0;
  padding: 0.5em 1em 0.75em !important;
}

.ui.card > .image:not(.ui) > img, .ui.cards > .card > .image:not(.ui) > img  {
  height: 212px;
}
