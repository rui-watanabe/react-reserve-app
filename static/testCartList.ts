import { CartProductType } from '../src/pagesTypes/CartType';
import { testProductList } from './testProductList';

export const testCartList: CartProductType = [
  {
    quantity: 1,
    product: testProductList[0],
  },
  {
    quantity: 2,
    product: testProductList[1],
  },
];
