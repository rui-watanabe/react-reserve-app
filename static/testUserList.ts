import { ObjectId } from 'mongodb';
import { UserModelType } from '../models/Types/UserType';

export const testUserList: UserModelType[] = [
  {
    _id: new ObjectId('000000000000000000000001'),
    createdAt: new Date('2020-9-22'),
    updatedAt: new Date('2020-9-22'),
    name: 'test',
    email: 'test@gmail.com',
    password: '11111111',
    role: 'root',
  },
  {
    _id: new ObjectId('000000000000000000000002'),
    createdAt: new Date('2020-9-22'),
    updatedAt: new Date('2020-9-22'),
    name: 'test2',
    email: 'test2@gmail.com',
    password: '22222222',
    role: 'admin',
  },
  {
    _id: new ObjectId('000000000000000000000003'),
    createdAt: new Date('2020-9-22'),
    updatedAt: new Date('2020-9-22'),
    name: 'test',
    email: 'test@gmail.com',
    password: '33333333',
    role: 'user',
  },
];
