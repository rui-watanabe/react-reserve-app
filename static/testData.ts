import { ProductModelType } from '../models/Types/ProductType';
import { UserModelType } from '../models/Types/UserType';

export const testProduct: ProductModelType = {
  _id: '11111',
  createdAt: new Date(),
  updatedAt: new Date(),
  name: 'Bernie Gledhill',
  price: 1399.02,
  description:
    'Morbi vestibulum, velit id pretium iaculis, diam erat fermentum justo, nec condimentum neque sapien placerat ante. Nulla justo. Aliquam quis turpis eget elit sodales scelerisque. Mauris sit amet eros. Suspendisse accumsan tortor quis turpis. Sed ante. Vivamus tortor. Duis mattis egestas metus.',
  sku: '594-54-9502',
  mediaUrl:
    'https://res-5.cloudinary.com/dwpujv6in/image/upload/c_pad,dpr_2.0,f_auto,h_930,q_auto,w_930/v1/media/catalog/product/f/d/fd1_lngchr_bh_frontlow-field-lounge-chair-tait-blush.jpg',
};

export const testProductList: ProductModelType[] = [testProduct];

export const testTotalPages = 1;

export const testUser: UserModelType = {
  _id: '00000000',
  createdAt: new Date(),
  updatedAt: new Date(),
  name: 'test',
  email: 'test@gmail.com',
  password: '11111111',
  role: 'user',
};
