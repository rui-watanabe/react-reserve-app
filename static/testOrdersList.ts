import { ObjectId, ObjectID } from 'mongodb';
import { OrderModelType } from '../models/Types/OrderType';
import { testProductList } from './testProductList';
import { testUserList } from './testUserList';

export const testOrdersList: OrderModelType[] = [
  {
    _id: new ObjectId('200000000000000000000001'),
    createdAt: new Date(2020, 4, 1, 12, 30, 45),
    updatedAt: new Date(2020, 4, 1, 12, 30, 45),
    products: [
      {
        _id: new ObjectID('100000000000000000000001'),
        quantity: 1,
        product: testProductList[0],
      },
      {
        _id: new ObjectID('100000000000000000000002'),
        quantity: 2,
        product: testProductList[1],
      },
    ],
    user: testUserList[2]._id, // 'user' role user
    email: testUserList[2].email,
    total: testProductList[0].price + testProductList[1].price * 2,
  },
  {
    _id: new ObjectID('200000000000000000000002'),
    createdAt: new Date(2020, 4, 1, 12, 30, 45),
    updatedAt: new Date(2020, 4, 1, 12, 30, 45),
    products: [
      {
        _id: new ObjectID('100000000000000000000003'),
        quantity: 1,
        product: testProductList[2],
      },
      {
        _id: new ObjectID('100000000000000000000004'),
        quantity: 2,
        product: testProductList[3],
      },
    ],
    user: testUserList[2]._id, // 'user' role user
    email: testUserList[2].email,
    total: testProductList[2].price + testProductList[3].price * 2,
  },
  {
    _id: new ObjectID('200000000000000000000003'),
    createdAt: new Date(2020, 4, 1, 12, 30, 45),
    updatedAt: new Date(2020, 4, 1, 12, 30, 45),
    products: [
      {
        _id: new ObjectID('100000000000000000000005'),
        quantity: 1,
        product: testProductList[4],
      },
      {
        _id: new ObjectID('100000000000000000000006'),
        quantity: 2,
        product: testProductList[5],
      },
    ],
    user: testUserList[2]._id, // 'user' role user
    email: testUserList[2].email,
    total: testProductList[4].price + testProductList[5].price * 2,
  },
];
