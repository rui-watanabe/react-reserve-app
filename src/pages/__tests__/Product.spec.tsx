import React from 'react';
import renderer from 'react-test-renderer';
import Product from '../product';
import { testUser, testProduct } from '../../../static/testData';

test('Product', () => {
  const component = renderer.create(
    <Product product={testProduct} user={testUser} />,
  );
  const tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});
