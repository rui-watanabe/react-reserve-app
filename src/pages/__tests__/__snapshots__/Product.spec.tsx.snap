// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product 1`] = `
Array [
  <div
    className="ui items"
  >
    <div
      className="item"
    >
      <div
        className="ui medium image"
      >
        <img
          src="https://res-5.cloudinary.com/dwpujv6in/image/upload/c_pad,dpr_2.0,f_auto,h_930,q_auto,w_930/v1/media/catalog/product/f/d/fd1_lngchr_bh_frontlow-field-lounge-chair-tait-blush.jpg"
        />
      </div>
      <div
        className="content"
      >
        <div
          className="header"
        >
          Bernie Gledhill
        </div>
        <div
          className="description"
        >
          <p>
            $
            1399.02
          </p>
          <div
            className="ui label"
            onClick={[Function]}
          >
            SKU: 
            594-54-9502
          </div>
        </div>
        <div
          className="extra"
        >
          <div
            className="ui action input"
          >
            <input
              min="1"
              onChange={[Function]}
              placeholder="Quantity"
              type="number"
              value={1}
            />
            <button
              className="ui orange button"
              onClick={[Function]}
            >
              <i
                aria-hidden="true"
                className="plus cart icon"
                onClick={[Function]}
              />
              Add to Cart
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>,
  <h3
    className="ui header"
  >
    About this product
  </h3>,
  <p>
    Morbi vestibulum, velit id pretium iaculis, diam erat fermentum justo, nec condimentum neque sapien placerat ante. Nulla justo. Aliquam quis turpis eget elit sodales scelerisque. Mauris sit amet eros. Suspendisse accumsan tortor quis turpis. Sed ante. Vivamus tortor. Duis mattis egestas metus.
  </p>,
]
`;
